### PR TITLE
Crypto migration task

### PR DESCRIPTION
**Pull request by roxiciobotea210**
Add task for changing encryption key in case it matches the historical one.
* get both keys & compare them
* if they match, generate a new key with the default algorithm
* update key settings
* migrate encrypted data to new key

### CODE REVIEW

**Review by jerome-benoit — CHANGES_REQUESTED** [review]
It should be a migration task, not a scheduler task. That's probably because I've not tell you where exactly to add it ;)
That's not difficult to make it a migration task. 
Otherwise looks good to me.

**Comment by roxiciobotea210** [comment]
> It should be a migration task, not a scheduler task. That's probably because I've not tell you where exactly add it ;)
> That's not difficult to make it a migration task.
> Otherwise looks good to me.

Ah, sorry, I created it following the example of the scheduler task Stefania created and that's why I added it here. Will change it to migration, thanks! :)

File: src/migration/tasks/ChangeCryptoKeyTask.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        crypto: {
+          formerKey: currentCryptoSettings.crypto.key,
+          formerKeyProperties: currentCryptoSettings.crypto.keyProperties,
+          key: Utils.generateRandomKey(Utils.getDefaultKeyProperties()),
+          keyProperties: Utils.getDefaultKeyProperties(),
+          migrationToBeDone: true
+        }
+      } as CryptoSettings;
+      await SettingStorage.saveCryptoSettings(tenant.id, keySettingToSave);
+      // migrate sensitive data to the new key
```
Comment (jerome-benoit) [inline src/migration/tasks/ChangeCryptoKeyTask.ts:43]: Nitpick: comment should start with an upper case letter.

**Review by jerome-benoit — APPROVED** [review]

**Review by LucasBrazi06 — APPROVED** [review]
GUT!

**Pull request merged**